api/internal/localizer/localizer.go

```diff
@@ -242,6 +244,48 @@ func (lc *localizer) localizeGenerator(generator *types.GeneratorArgs) (err erro
 	return nil
 }
 
+// localizeHelm localizes helm values files and copies local chart homes.
+func (lc *localizer) localizeHelm(kust *types.Kustomization) error {
+	for i, legacy := range kust.HelmChartInflationGenerator {
+		locFile, err := lc.localizeFileIfSet(legacy.Values)
+		if err != nil {
+			return errors.WrapPrefixf(err, "unable to localize helmChartInflationGenerator entry %d values", i)
+		}
+		kust.HelmChartInflationGenerator[i].Values = locFile
+
+		if legacy.ChartHome != "" {
+			locDir, err := lc.copyChartHome(legacy.ChartHome)
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to copy helmChartInflationGenerator entry %d", i)
+			}
+			kust.HelmChartInflationGenerator[i].ChartHome = locDir
+		}
+	}
+	for i, chart := range kust.HelmCharts {
+		locFile, err := lc.localizeFileIfSet(chart.ValuesFile)
+		if err != nil {
+			return errors.WrapPrefixf(err, "unable to localize helmCharts entry %d valuesFile", i)
+		}
+		kust.HelmCharts[i].ValuesFile = locFile
+	}
+	if kust.HelmGlobals != nil {
+		if kust.HelmGlobals.ChartHome != "" {
+			locDir, err := lc.copyChartHome(kust.HelmGlobals.ChartHome)
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to copy helmGlobals")
+			}
+			kust.HelmGlobals.ChartHome = locDir
+		}
+	}
+	if useDefaultChartHome(kust) {
```

Comment:
Do you need to do this separately? It seems like you can just fold the logic into `copyChartHome`:

```
func (lc *localizer) copyChartHome(path string) (string, error) {
  if path == "" {
    path = defaultChartHome
    // some sort of check to see if you've already copied the default chart home, 
    // probably just by checking if the copied directory already exists      
  }
  ...
}
```

If we can, I think this would be a better structure because it keeps the chartHome code together and stops us from having to iterate over the legacy helm charts twice.

Reply:
Done. I folded the empty path check into `copyChartHome`. I added a field to keep track of whether we have copied the default chart home: https://github.com/kubernetes-sigs/kustomize/pull/4978/files#diff-25faa786bf35469d391d5eef99ed1323623023442b177c1ed3e07f3e6709dac6R39-R41
`charts` could exist at the localized copy without the entirety of its contents if a kustomization field (maybe a helm field) referenced a file in `charts`. If we don't care about this edge case, I can remove the field I added and simply check if `charts` exists, like you recommended.

However, there is still code checking if chart home is empty:
* https://github.com/kubernetes-sigs/kustomize/pull/4978/files#diff-25faa786bf35469d391d5eef99ed1323623023442b177c1ed3e07f3e6709dac6R262-R264
* https://github.com/kubernetes-sigs/kustomize/pull/4978/files#diff-25faa786bf35469d391d5eef99ed1323623023442b177c1ed3e07f3e6709dac6R279-R290
because we want to preserve empty chart homes in the original kustomizations. If we don't care about this preservation, I can remove these checks.

Reply:
I updated `copyChartHome` to return an empty string if it's passed an empty string.